code/osapi/osapi.cpp

```diff
@@ -318,19 +321,31 @@ bool os_is_legacy_mode()
 		checkedLegacyMode = true;
 	}
 	else {
+		bool old_config_exists = false;
+		bool new_config_exists = false;
+
 		SCP_stringstream path_stream;
 		path_stream << getPreferencesPath() << DIR_SEPARATOR_CHAR << Osreg_config_file_name;
 
-		// Use the existance of the fs2_open.ini file for determining if the launcher supports the new mode
-		auto file = fopen(path_stream.str().c_str(), "r");
+		new_config_exists = file_exists(path_stream.str());
+#ifdef SCP_UNIX
+        path_stream.str("");
+		path_stream << Cfile_user_dir_legacy << DIR_SEPARATOR_CHAR << Osreg_config_file_name;
 
-		if (file == nullptr)
-		{
+		old_config_exists = file_exists(path_stream.str());
+#else
+		// At this point we can't determine if the old config exists so just assume that it does
```

Comment:
Well, that obviously won't work because `os_init()` checks various command-line flags. In fact, `os_is_legacy_mode()` and `os_get_config_path()` both check `Cmdline_portable_mode`, which is useless since they both get called before the command line is actually parsed (unless "-parse_cmdline_only" is present). Hrmph... clearly this is a deeper problem that requires some additional thought.